benchmarks/rayleigh_taylor_instability/rayleigh_taylor_instability.cc

```diff
@@ -0,0 +1,236 @@
+                                 "Viscosity in the Inclusion.");
+              prm.declare_entry ("rho1", "3200",
+                                 Patterns::Double (0),
+                                 "density of the mantle.");
+              prm.declare_entry ("rho2", "3232",
+                                 Patterns::Double (0),
+                                 "density in the Inclusion.");
+              prm.declare_entry ("amplitude", "1000",
+                                 Patterns::Double (0),
+                                 "perturbation amplitude.");
+              prm.declare_entry ("lambda", "256e3",
+                                 Patterns::Double (0),
+                                 "perturbation wavelength.");
+            }
+            prm.leave_subsection();
+          }
+          prm.leave_subsection();
+        }
+
+        /**
+         * Read the parameters this class declares from the parameter file.
+         */
+        virtual
+        void
+        parse_parameters (ParameterHandler &prm)
+        {
+          prm.enter_subsection("Material model");
+          {
+            prm.enter_subsection("RTinstability");
+            {
+              eta1 = prm.get_double ("eta1");
+              eta2 = prm.get_double ("eta2");
+              rho1 = prm.get_double ("rho1");
+              rho2 = prm.get_double ("rho2");
+              amplitude = prm.get_double ("amplitude");
+              lambda = prm.get_double ("lambda");
+            }
+            prm.leave_subsection();
+          }
+          prm.leave_subsection();
+
+          // Declare dependencies on solution variables
+          this->model_dependence.viscosity = MaterialModel::NonlinearDependence::none;
+          this->model_dependence.density = MaterialModel::NonlinearDependence::none;
+          this->model_dependence.compressibility = MaterialModel::NonlinearDependence::none;
+          this->model_dependence.specific_heat = MaterialModel::NonlinearDependence::none;
+          this->model_dependence.thermal_conductivity = MaterialModel::NonlinearDependence::none;
+        }
+
+        /**
+         * @name Reference quantities
+         * @{
+         */
+        virtual double reference_viscosity () const;
+        /**
+         * @}
+         */
+
+      private:
+        double eta1;
+        double eta2;
+        double rho1;
+        double rho2;
+        double amplitude;
+        double lambda;
+
+    };
+
+
+    template <int dim>
+    double
+    RTinstabilityMaterial<dim>::
+    reference_viscosity () const
+    {
+      return 1.e21;
+    }
+
+
+    template <int dim>
+    bool
+    RTinstabilityMaterial<dim>::
+    is_compressible () const
+    {
+      return false;
+    }
+
+    /**
+      * A postprocessor that evaluates the accuracy of the solution.
+      *
+      * The implementation of error evaluators that correspond to the
+      * benchmarks defined in the Donea and Huerta FEM book (see manual).
```

Comment:
Is this copy-pasted from a different benchmark? Your comment suggests this one is not from the Donea and Huerta book?